pygeos/_geometry.pyx

```diff
@@ -69,9 +183,13 @@ def get_parts(object[:] array):
             for part_idx in range(counts_view[geom_idx]):
                 index_view[idx] = geom_idx
                 part = GEOSGetGeometryN_r(geos_handle, geom, part_idx)
+                if part == NULL:
```

Comment:
This `return` doesn't seem right; the expected return shape is a tuple of ndarrays.

This should raise an exception instead: we expected a part and we didn't get one back.  The GEOS docs state that a `NULL` return here is an exception.

Reply:
If you do `return NULL` from C, you tell python to raise an exception. It will raise an exception reading something like "function returned NULL without an exception message set".

However, we do set an exception message. We basically trust GEOS here to have called the `setErrorMessageHandler` which result in a `GEOSException` in the `get_geos_handle().__exit__`. It seems a bit convolved, but it is the only way I know to be able to catch the GEOS exception message. I'll make a comment on the `return NULL`.